client/allocrunnerv2/alloc_runner.go

```diff
@@ -428,7 +428,15 @@ func (ar *allocRunner) Destroy() {
 	}
 
 	// Wait for tasks to exit and postrun hooks to finish
-	<-ar.waitCh
+WAIT:
+	for {
+		select {
+		case <-ar.waitCh:
+			break WAIT
+		case <-time.After(time.Minute):
+			ar.logger.Info("waiting for tasks to exit")
```

Comment:
Maybe add a field for remaining tasks?

Reply:
Funny story! The Kill() calls above are synchronous and therefore this log message is wrong. It only appears if the _postrun_ hooks block or AR.Run doesn't exit for some other reason... Regardless I think at this point it's only due to a bug that we wouldn't exit in a timely matter.

I'm going to remove this log line for now since I can't think of anything sensible to log. If we hit another deadlock on `ar.waitCh` it should either lead us to the bug or to something sensible to log.